src/fast_check/transform_dts.rs

```diff
@@ -0,0 +1,730 @@
+use deno_ast::swc::{
+  ast::{
+    BindingIdent, ClassMember, Decl, DefaultDecl, ExportDecl,
+    ExportDefaultExpr, Expr, Ident, Lit, MethodKind, Module, ModuleDecl,
+    ModuleItem, Pat, Prop, PropName, PropOrSpread, Stmt, TsFnOrConstructorType,
+    TsFnParam, TsFnType, TsPropertySignature, TsTupleElement, TsTupleType,
+    TsType, TsTypeElement, TsTypeLit, VarDecl, VarDeclKind, VarDeclarator,
+  },
+  common::DUMMY_SP,
+};
+
+use crate::FastCheckDiagnostic;
+
+use super::swc_helpers::{
+  any_type_ann, maybe_lit_to_ts_type, maybe_lit_to_ts_type_const, ts_readonly,
+  ts_tuple_element, type_ann,
+};
+
+pub struct FastCheckDtsTransformer {
+  id_counter: usize,
+}
+
+impl FastCheckDtsTransformer {
+  pub fn new() -> Self {
+    Self { id_counter: 0 }
+  }
+
+  fn gen_unique_name(&mut self) -> String {
+    self.id_counter += 1;
+    format!("_dts_{}", self.id_counter)
+  }
+
+  pub fn transform(
+    &mut self,
+    mut module: Module,
+  ) -> Result<Module, Vec<FastCheckDiagnostic>> {
+    let mut body = module.body;
+
+    let mut new_items: Vec<ModuleItem> = vec![];
```

Comment:
We can pre-allocate this:

```suggestion
    let mut new_items: Vec<ModuleItem> = Vec::with_capacity(body.len());
```